tools/test-management

```diff
@@ -1,18 +1,58 @@
-#!/bin/bash
-set -e
-
-echo "Testing management commands start!"
-failed=0
-for i in `ls */management/commands/ | grep .py$ | grep -v __init__ | sed 's/[.]py$//'`; do
-    if ! $(./manage.py $i --help >/dev/null); then
-        failed=$(expr $failed + 1)
-        echo "ERROR: \`./manage.py $i --help\` crashes!";
-    fi
-done
-
-if [ $failed -gt 0 ]; then
-   echo
-   echo "$failed management commands don't start!"
-   exit 1
-fi
-echo "All management commands start!"
+#!/usr/bin/env python
+
+from __future__ import absolute_import, print_function
+
+import os
+import sys
+import glob
+from importlib import import_module
+
+import django
+from django.apps import apps
+from django.core import management
+from django.utils.module_loading import module_has_submodule
+
+sys.path.insert(0, os.path.dirname('..'))
+os.environ['DJANGO_SETTINGS_MODULE'] = 'zproject.test_settings'
+django.setup()
+
+for app_config in apps.get_app_configs():
+    if module_has_submodule(app_config.module, 'management'):
+        import_module('.management', app_config.name)
+
+commands = filter(
+    lambda filename: filename != '__init__',
+    map(
+        lambda file: os.path.basename(file).replace('.py', ''),
+        glob.iglob('*/management/commands/*.py')
+    )
+)
+
+failed_commands = []
+
+print('Testing management commands start!')
+
+original_stdout = sys.stdout
+sys.stdout = open(os.devnull, 'a')
+
+for command in commands:
+    try:
+        management.call_command(command, '--help')
+
+    except SystemExit as e:
+        # '--help' return exit status 0.
+        if str(e) == "0":
+            pass
+
+    except Exception as e:
+        print(e, file=original_stdout)
```

Comment:
We probably want `traceback.print_exc(e)` to show the full exception.  Also maybe worth printing the specific command that failed here on a line before the traceback.